README.md

```diff
@@ -1,2 +1,373 @@
-# steampipe-action
-This Github Action runs Steampipe against your cloud infrastructure. Use SQL to instantly query your cloud services (AWS, Azure, GCP and more). Open source CLI. No DB required.
+# Steampipe Check Action for GitHub Actions
+
+This GitHub Action runs [Steampipe](https://steampipe.io/) Checks (benchmarks/controls) from [Steampipe Mods](https://hub.steampipe.io/mods).
+
+This action will produce an easy to read **Job Summary** as part of your workflow.
+
+<img src="images/example_summary.png" width="80%">
+
+For a full list of configuration options and descriptions see [action.yml](action.yml).
+
+## IaC Checks & Annotations
+
+This mod also allows you to scan your Infrastructure as Code (IaC) files directly from your GitHub repository using your workflow pipeline. This helps to identify potential security vulnerabilities, compliance issues, and infrastructure misconfigurations early in the development cycle.
+
+For controls that scan local files, like those in the [Terraform AWS Compliance mod](https://github.com/turbot/steampipe-mod-terraform-aws-compliance), annotations will be created for any controls in `alarm` in the workflow provided they have an [additional dimension](https://steampipe.io/docs/reference/mod-resources/control#additional-control-columns--dimensions) providing the file path and a start (& optionally end) line numbers in the following formats: `path/filename.ext:23` or `path/name.ext:23-25`.
+
+<img src="images/example_annotations.png" width="80%" />
+
+For a list of IaC mods offered by Turbot, please see [IaC mods](https://hub.steampipe.io/mods?categories=iac)
+
+> NOTE: In order to create the annotations, you may need to ensure that the GitHub token provided has the permissions to write under `Settings -> Actions -> Workflow Permissions`, alternatively you can pass the `pull-requests: write` and `checks: write` permissions to the `job` see [job permissions](https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs).
+
+## Usage
+
+> This action requires you set-up Steampipe in your workflow in advance of using it, the recommend approach would be to utilise the [turbot/steampipe-action-setup](https://github.com/turbot/steampipe-action-setup) action, however you can set this up manually if you prefer.
+
+### Basic Example
```

Comment:
Can you please add 2 examples:
- One setting up a GitHub plugin connection (can probably pass the token in as part of the GitHub plugin connection) and then run some smaller benchmarks or controls from the GitHub Compliance mod
- Run some AWS CIS v2.0.0 benchmarks with AWS creds setup using [OIDC](https://aws.amazon.com/blogs/security/use-iam-roles-to-connect-github-actions-to-actions-in-aws/). We should link to the AWS blog and provide a sample workflow showing how to use the `aws-actions/configure-aws-credentials` action, installing the AWS plugin, and then running some of the checks.

These will help highlight how this action can be used for any mod check, not just IaC mods

Reply:
GH example added.

Just reading through the OIDC thing to establish how this would work.

Reply:
Added AWS OIDC example